eng/tools/spec-gen-sdk-runner/src/command-helpers.ts

```diff
@@ -264,6 +264,7 @@ export function generateArtifact(
       path.join(commandInput.workingFolder, specGenSdkArtifactPath, specGenSdkArtifactFileName),
       JSON.stringify(artifactInfo, undefined, 2),
     );
+    setVsoVariable("SpecGenSdkServiceNames", commandInput.serviceNames!);
```

Comment:
What does the `!` at the end represent?

Reply:
Assigning `string | undefined` to `string`.